### DISCUSSION TITLE
How to alter transcription result in Whisper?

### DISCUSSION THREAD

**Discussion by AbahAya** [openai/whisper#1626] [Q&A]
Hello everyone!

Question 1: 
**Are there commands for Whisper to alter its transcription results?**
I need Whisper to replace all the punctuation for the end of a sentence, such as full stop (.), acclamation (!), or question mark (?) to be written as a distinctive new marking.

Question 2:
**Is there any command for Whisper to alter the writing layout of its transcript result?** 

This is an example of an SRT file of the Whisper transcript result:

1
00:00:00,000 --> 00:00:15,000
Nadim tuh bagian awal dari eksperimen ide kita dengan Endgame. Kalau ada yang sempet lihat, di bulan September tahun 2019 itu sempet ngobrol dengan Nadim atau Mas Menteri.

This is what I want the Whisper to make:

        intervals [1]:
            xmin = 0
            xmax = 15 
            text = "nadiem [tuh] bagian awal dari <en> experiment </en> ide kita dengan <en> end game </en> <s/> kalau ada yang sempat lihat <c/> di bulan september tahun dua ribu sembilan belas saya [tuh] sempat ngobrol dengan nadiem atau mas menteri <s/>"

Question 3:
**Is there any command to let Whisper transcribe words exactly as it is including the particles?**
Apparently, Whisper did the job too well.

I want Whisper to write like this:
"...and then, we can pe~ pe~ perceive it as s~ so~ something else."
instead of
"...and then, we can perceive it as something else."

I really appreciate any help you can provide.

**Comment by glangford** [comment]
Question 1: whisper will not separate punctuation, you need to post-process the output and do this yourself (perhaps with regular expressions or an NLP tool). See also

* https://github.com/openai/whisper/discussions/589

Question 2: whisper does not support changing languages like the example you give, but it's a feature that has been requested before.

* https://github.com/openai/whisper/discussions/345
* https://github.com/openai/whisper/discussions/49
* https://github.com/openai/whisper/discussions/1318

Question 3: See
* https://github.com/openai/whisper/discussions/286
* https://github.com/openai/whisper/discussions/1517
